src/Umbraco.Cms.Api.Management/Extensions/TypeExtensions.cs

```diff
@@ -0,0 +1,12 @@
+using Umbraco.Cms.Api.Management.Controllers;
+
+namespace Umbraco.Cms.Api.Management.Extensions;
+
+public static class TypeExtensions
+{
+    public static bool IsManagementApiController(this Type type)
+        => typeof(ManagementApiControllerBase).IsAssignableFrom(type);
+
+    public static bool IsManagementApiViewModel(this Type type)
+        => type.Namespace?.StartsWith("Umbraco.Cms.Api.Management.ViewModels") is true;
```

Comment:
I haven't tested this, but if I'm reading the code correctly, this effectively means any custom APIs would not yield the same error casing as core ones?

We've made some efforts to enable parity between custom APIs and core APIs - for example offering to apply the [same schema and operation IDs](https://docs.umbraco.com/umbraco-cms/tutorials/creating-a-backoffice-api/umbraco-schema-and-operation-ids). I wonder if we really want to exclude custom APIs from this casing fix?

Reply:
That is correct, but I do not see a way of gracefully handling this within our logic besides relying on an attribute on the (custom) models or using some kind of inheritance.

Or do you want to including something similar to 
```csharp
public class SampleSchemaIdHandler : SchemaIdHandler
{
    public override bool CanHandle(Type type)
        => type.Namespace == "UmbracoDocs.Samples";
}
```
and then in `ManagementApiSystemTextJsonValidationMetadataProvider.CreateValidationMetadata()` loop over all Handlers (not what I would call it) to see if any specify that namespace and process it as if it were a management api model?

Reply:
Having mulled this over a little bit more, I would actually expect the validation casing to be (forcefully) applied to my API if I've explicitly built on top of the `ManagementApiControllerBase` - this would ensure consistency in error handling for any Management API consumers?

Reply:
I agree. Sadly you can not reliably get this information from every class/property that flows trough that method, at least not according to my experimentation and understanding of the docs.